WalletWasabi.Daemon/Config.cs

```diff
@@ -168,4 +146,80 @@ public Uri GetCoordinatorUri()
 
 		return result is null ? GetBackendUri() : new Uri(result);
 	}
+
+	private static string GetString(string valueInConfigFile, string[] args, string key)
+	{
+		return GetEffectiveValue(valueInConfigFile, x => x, args, key);
+	}
+
+	private static T GetEffectiveValue<T>(T valueInConfigFile, Func<string, T> converter, string[] args, string key)
+	{
+		if (ArgumentHelpers.TryGetValue(key, args, converter, out var cliArg))
+		{
+			return cliArg;
+		}
+
+		var envKey = "WASABI-" + key.ToUpperInvariant();
+		var envVars = Environment.GetEnvironmentVariables();
+		if (envVars.Contains(envKey))
+		{
+			if (envVars[envKey] is string envVar)
+			{
+				return converter(envVar);
+			}
+		}
+
+		return valueInConfigFile;
+	}
+
+	private EndPoint GetEffectiveEndPoint(EndPoint valueInConfigFile, string key)
+	{
+		return GetEffectiveValue(
+			valueInConfigFile,
+			x =>
+			{
+				if (EndPointParser.TryParse(x, 0, out var endpoint))
+				{
+					return endpoint;
+				}
+
+				throw new ArgumentNullException(key, "Not a valid endpoint");
+			},
+			Args,
+			key);
+	}
+
+	private bool GetEffectiveBool(bool valueInConfigFile, string key)
+	{
+		return GetEffectiveValue(
+			valueInConfigFile,
+			x =>
+				bool.TryParse(x, out var value)
+					? value
+					: throw new ArgumentException("must be 'true' or 'false'.", key),
+			Args,
+			key);
+	}
+
+	private string GetEffectiveString(string valueInConfigFile, string key)
+	{
+		return GetEffectiveValue(valueInConfigFile, x => x, Args, key);
+	}
+
+	private string? GetEffectiveOptionalString(string? valueInConfigFile, string key)
+	{
+		return GetEffectiveValue(valueInConfigFile, x => x, Args, key);
+	}
+
+	private T GetEffectiveValue<T>(T valueInConfigFile, Func<string, T> converter, string key)
+	{
+		var effectiveValue = GetEffectiveValue(valueInConfigFile, converter, Args, key);
+
+		if (!Equals(effectiveValue, valueInConfigFile))
+		{
+			IsOverridden = true;
+		}
+
+		return effectiveValue;
```

Comment:
Shouldn't all `GetEffectiveValue` functions do a check and set `IsOverride` if needed? Right now only 3 parameters- `Network`, `DustThreshold` and `JsonRpcServerPrefixes` -will be aware that there was a change.

Reply:
🙈 You're absolutely right.
Fixed.

Please, notice how there's still a static call that can't be handled (`GetString`). We don't want IsOverridden to be static, for obvious reasons.